scripts/samba-vuln-cve-2017-7494.nse

```diff
@@ -0,0 +1,531 @@
+--  writable share and its path
+local function find_writable_shares(host, samba_cve)
+  -- determine if there are writable shares
+  local status, main_name, main_path, names
+  status, main_name, main_path, names = smb.share_find_writable(host)
+
+  -- successful in finding writable share
+  if status then
+    local msg = string.format("Writable share found. \n Name: %s", main_name)
+    if main_path then
+      msg = msg .. string.format("\n Path: %s ", main_path)
+    end
+
+    -- insert main writable directory with path into check_results
+    table.insert(samba_cve.check_results, msg)
+
+    -- insert names of other writable shares to extra_info
+    if #names > 0 then
+      table.insert(samba_cve.extra_info, string.format(
+        "All writable shares:"))
+    end
+    for i = 1, #names, 1 do
+      table.insert(samba_cve.extra_info, string.format(" Name: %s", main_name))
+    end
+  else
+    -- writable share enumeration failed, return error message stored in main_name
+    local err = main_name
+    table.insert(samba_cve.extra_info, err)
+    main_name = nil
+  end
+
+  -- main_path is C:\<actual share>
+  -- we map it to the equivalent statement in Unix filesystems
+  -- i.e. /<actual share>/
+  if main_path then
+    main_path = "/" .. string.sub(main_path, 4) .. "/"
+  end
+
+  return main_name, main_path
+end
+
+--- Check if the suggested workaround "nt pipe support = no" was applied on
+--  the target host. The script checks if details can be queried on IPC$
+--  which in a typical case will return details on the IPC, but if the
+--  workaround is applied, an error of 'NT_STATUS_ACCESS_DENIED' is returned
+--
+--  @param host The target host
+--  @param samba_cve The vuln table containing information for the results
+--  @return A boolean indicating the nt pipe support is enabled, which
+--          indicates the workaround was not applied
+local function is_ntpipesupport_enabled(host, samba_cve)
+  -- do "nt pipe support = no" workaround check, in which case
+  -- accessing 'IPC$' returns 'NT_STATUS_ACCESS_DENIED'
+  local status, result
+  status, result = smb.share_get_details(host, 'IPC$')
+
+  if status and result['details'] == "NT_STATUS_ACCESS_DENIED" then
+    samba_cve.state = vulns.STATE.NOT_VULN
+    return false
+  elseif not status then
+    -- error accessing IPC$, present error to user
+    local err = result
+    table.insert(samba_cve.extra_info, err)
+  end
+
+  return true
+end
+
+--- Creates candidate paths for common directories of shares
+--  This is method is based off the Metasploit script.
+--
+--  @param share_name Name of the share that you wish to write to
+--  @return Array of candidate paths of the shares, never nil
+local function enumerate_directories(share_name)
+  local common_dirs = {"/volume1/","/volume2/","/volume3/","/volume4/",
+    "/shared/","/mnt/","/mnt/usb/","/media/","/mnt/media/","/var/samba/",
+    "/tmp/","/home/","/home/shared/"}
```

Comment:
I would move the `common_dirs` along with the other constants (payloads). An advanced user may want to edit the script and add more dirs here.